rtos/TARGET_CORTEX/TARGET_CORTEX_M/mbed_rtx_fault_handler.c

```diff
@@ -17,278 +17,117 @@
+void mbed_fault_handler (uint32_t fault_type, void *mbed_fault_context_in, void *osRtxInfoIn)
 {
     mbed_error_status_t faultStatus = MBED_SUCCESS;
     
-    fault_print("\n++ MbedOS Fault Handler ++\n\nFaultType: ",NULL);
+    mbed_error_printf("\n++ MbedOS Fault Handler ++\n\nFaultType: ");
         
     switch( fault_type ) {
-      case HARD_FAULT_EXCEPTION: 
-        fault_print("HardFault",NULL);
-        faultStatus = MBED_ERROR_HARDFAULT_EXCEPTION;      
-        break;
       case MEMMANAGE_FAULT_EXCEPTION: 
-        fault_print("MemManageFault",NULL); 
+        mbed_error_printf("MemManageFault"); 
         faultStatus = MBED_ERROR_MEMMANAGE_EXCEPTION;
         break;
+      
       case BUS_FAULT_EXCEPTION: 
-        fault_print("BusFault",NULL); 
+        mbed_error_printf("BusFault"); 
         faultStatus = MBED_ERROR_BUSFAULT_EXCEPTION;
         break;
+      
       case USAGE_FAULT_EXCEPTION: 
-        fault_print("UsageFault",NULL); 
+        mbed_error_printf("UsageFault"); 
         faultStatus = MBED_ERROR_USAGEFAULT_EXCEPTION;
         break;
-      default: 
-        fault_print("Unknown Fault",NULL); 
-        faultStatus = MBED_ERROR_UNKNOWN;
+      
+      //There is no way we can hit this code without getting an exception, so we have the default treated like hardfault
+      case HARD_FAULT_EXCEPTION: 
+      default:    
+        mbed_error_printf("HardFault");
+        faultStatus = MBED_ERROR_HARDFAULT_EXCEPTION;      
         break;
     }
-    fault_print("\n\nContext:",NULL);
+    mbed_error_printf("\n\nContext:");
     print_context_info();
-        
-    fault_print("\n\nThreads Info:\nCurrent:",NULL);
-    print_thread(((osRtxInfo_t *)osRtxInfoIn)->thread.run.curr);
-  
-    fault_print("\nNext:",NULL);
-    print_thread(((osRtxInfo_t *)osRtxInfoIn)->thread.run.next);
-    
-    fault_print("\nWait:",NULL);
-    osRtxThread_t *threads = ((osRtxInfo_t *)osRtxInfoIn)->thread.wait_list;
-    print_threads_info(threads);
-    
-    fault_print("\nDelay:",NULL);
-    threads = ((osRtxInfo_t *)osRtxInfoIn)->thread.delay_list;
-    print_threads_info(threads);
     
-    fault_print("\nIdle:",NULL);
-    threads = ((osRtxInfo_t *)osRtxInfoIn)->thread.idle;
-    print_threads_info(threads);
-    
-    fault_print("\n\n-- MbedOS Fault Handler --\n\n",NULL);
+    mbed_error_printf("\n\n-- MbedOS Fault Handler --\n\n");
     
     //Now call mbed_error, to log the error and halt the system
-    mbed_error( MBED_MAKE_ERROR( MBED_MODULE_UNKNOWN, faultStatus ), "System encountered an unrecoverable fault excaption, halting system.", mbed_fault_context.PC_reg, NULL, 0 );
+    mbed_error( faultStatus, "Unrecoverable fault excaption.", mbed_fault_context.PC_reg, NULL, 0 );
     
-    /* In case we return, just spin here, we have already crashed */
-    for (;;) { 
-        __WFI(); 
-    }
 }
 
-void print_context_info(void)
+MBED_NOINLINE void print_context_info(void)
```

Comment:
inlining was causing some problems for this function?

Reply:
No specific problems with inlining. But sometimes debugging and also identifying the space required for individual functions becomes hard. So I removed inline as I was working on some optimizations.